app/gui/src/presenter/searcher.rs

```diff
@@ -280,15 +281,19 @@ impl Searcher {
                     new_input <- list_view.suggestion_accepted.filter_map(f!((e) model.suggestion_accepted(*e)));
                     graph.set_node_expression <+ new_input;
 
-                    entry_selected <- list_view.selected_entry
-                        .map2(&list_view.is_header_selected, |entry, (_, is_header)| (!is_header).as_some(*entry))
-                        .filter_map(|e| *e);
+                    entry_selected <- list_view.selected.map(|s| match s {
+                        Some(list_panel::Selected::Entry(e)) => Some(*e),
+                        _ => None
+                    });
```

Comment:
Linter should hint that this is the same as `.map`. Was the code linted? If so, could you check the linter configuration and see why it was not hinted here, please? The same applies to the code below. In either case, reply to this comment what happened here, please. https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0a8154155b3cb64fbc8b7378ce37a59b

Reply:
Your example does not resemble the actual state of the code. [This one does](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8caa79f5e9a0f2cac2cedf4cac530205). This code uses the pattern matching to filter a specific enum variant, so it can't be expressed as `.map` call.

There is a way to rewrite it as `list_view.selected.map(|s| s.map(list_panel::Selected::as_entry))`, but it would require adding `as_entry` method which will do the similar pattern matching inside. I guess I can go that road.

Reply:
Update: I implemented the `as_entry_id` method as part of the refactoring. The other similar match below doesn't worth it, I think. (or I somewhat can't find a good name for such a method)